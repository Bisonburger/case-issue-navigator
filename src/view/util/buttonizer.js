import React from "react";
import UsaButton from "./UsaButton";

const buttonizer = (text, buttonClass, callbackKey) => {
  return (_, rowData, __, callback) => (
    <UsaButton
      onClick={() => callback[callbackKey](rowData)}
      buttonStyle={buttonClass}
    >
      {text}
    </UsaButton>
  );
};

export { buttonizer };
