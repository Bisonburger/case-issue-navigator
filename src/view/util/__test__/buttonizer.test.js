import { shallow } from "enzyme";
import { buttonizer } from "../buttonizer";

describe("buttonizer", () => {
  const myFunc = jest.fn(x => x);
  const callback = {
    myFunc: rowdata => myFunc(rowdata)
  };
  const rowdata = {
    receiptNumber: "FAK123"
  };
  const buttonized = buttonizer("Click Me", "outline", "myFunc");

  it("renders a button", () => {
    const Button = buttonized(null, rowdata, null, callback);
    expect(Button).toMatchSnapshot();
  });
  it("executes callback when clicked", () => {
    const Button = shallow(buttonized(null, rowdata, null, callback));
    Button.simulate("click");
    expect(myFunc).toHaveBeenCalledTimes(1);
  });
});
