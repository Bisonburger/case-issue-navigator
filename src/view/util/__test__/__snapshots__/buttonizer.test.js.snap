// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttonizer renders a button 1`] = `
<UsaButton
  buttonStyle="outline"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Click Me
</UsaButton>
`;
